### PR TITLE
Fix warnings found by luacheck

### LINKED ISSUE #208: Check Lua source code with luacheck
see also [Add a lua linter check step before push to repo #4681](https://github.com/tarantool/tarantool/issues/4681)

### PR DESCRIPTION
**Pull request by ligurio**
Closes #208

### CODE REVIEW

File: test_run.lua [hunk 1]
```diff
@@ -102,7 +102,7 @@ local create_cluster_cmd2 = 'start server %s with wait_load=False, wait=False'
 local create_cluster_cmd2_args = create_cluster_cmd2 .. ', args="%s"'
 
 local function create_cluster(self, servers, test_suite, opts)
-    local opts = opts or {}
+    opts = opts or {}
```
Comment (Totktonada) [inline test_run.lua:105]: As already [said](https://lists.tarantool.org/pipermail/tarantool-patches/2020-June/017308.html), I would not forbid redefinitions / hiding variables. Can we place a .luacheckrc file into the repository root to achieve this?

File: test_run.lua [hunk 2]
```diff
@@ -163,15 +163,15 @@ local function wait_fullmesh(self, servers)
                 log.info("%s: bootstrapped", server)
                 break
             end
-            local info = self:eval(server, "box.info")
+            self:eval(server, "box.info")
```
Comment (Totktonada) [inline test_run.lua:166]: To be honest, I don't understand why `test_run:eval() `it is called here.

File: test_run.lua [hunk 3]
```diff
@@ -163,15 +163,15 @@ local function wait_fullmesh(self, servers)
...
             fiber.sleep(0.01)
         end
         -- wait all for full mesh
         for _, server2 in ipairs(servers) do
             if server ~= server2 then
                 log.info("%s -> %s: waiting for connection", server2, server)
                 while true do
-                    local cmd = wait_fullmesh_cmd:format(server_id)
+                    wait_fullmesh_cmd:format(server_id)
```
Comment (Totktonada) [inline test_run.lua:174]: `string.format()` returns a new string: it does not change a string in place. Strings in Lua are immutable.

File: test_run.lua (hunk 1 above)
Comment (ligurio) [reply test_run.lua:105]: My bad, I have pushed changes made before our discussion. New patch contains reverted redefinitions back.

File: test_run.lua (hunk 2 above)
Comment (ligurio) [reply test_run.lua:166]: Me too. Just removed line and comment it in a commit message.

File: test_run.lua (hunk 3 above)
Comment (ligurio) [reply test_run.lua]: Reverted change.

File: test_run.lua (hunk 2 above)
Comment (Totktonada) [reply test_run.lua]: The line was here from the beginning (48553b239420166e7f7c26dc01f0ec08c36778d6) and it does not look meaningful within the original code too. Looks as a mistake. So I'm okay with removing if tarantool testing works after the change.

**Review by Totktonada — APPROVED** [review]
The changes look okay for me. I'll test it a bit later and if everything will be okay, will push and update the submodule in tarantool.

**Comment by ligurio** [comment]
> The changes look okay for me. I'll test it a bit later and if everything will be okay, will push and update the submodule in tarantool.

tests passed with tarantool built from master

**Comment by Totktonada** [comment]
Thanks!

Updated the test-run submodule in tarantool in [2.7.0-40-g546499c9c][1], [2.6.1-30-g7f2eb13f8][2], [2.5.2-19-gc91b1e568][3], [1.10.8-16-g0ef495879][4].

[1]: https://github.com/tarantool/tarantool/commit/546499c9c001e30cbd2598946c9c0589b8e30f53
[2]: https://github.com/tarantool/tarantool/commit/7f2eb13f83ec1f1068d9d1d90d81745a52c94fb9
[3]: https://github.com/tarantool/tarantool/commit/c91b1e568cf77ef7d912d832127aa38e755d0d19
[4]: https://github.com/tarantool/tarantool/commit/0ef4958792a9be2f3d92f9b6cf56a5ff1a3584d7

**Pull request merged**